app/service/plugin/HookManager.php

```diff
@@ -114,10 +115,18 @@ public function doAction(string $hook, mixed ...$args): void
                 }
                 // once 支持：约定传入的回调若为一次性，执行后移除
                 if (is_array($item) && ($item['_once'] ?? false) === true) {
-                    unset($list[$idx]);
-                    $this->actions[$hook][$priority] = array_values($list);
+                    $toRemove[] = $idx;
                 }
             }
+            // 批量移除一次性钩子
+            if (!empty($toRemove)) {
```

Comment:
在HookManager中改进了一次性钩子的移除机制，从直接在遍历中修改数组改为收集需要移除的索引，然后批量移除。这是一个很好的改进，避免了在遍历过程中修改数组可能引发的问题。

```suggestion
// 批量移除一次性钩子
if (!empty($toRemove)) {
    foreach ($toRemove as $idx) {
        unset($this->actions[$hook][$priority][$idx]);
    }
    $this->actions[$hook][$priority] = array_values($this->actions[$hook][$priority]);
    // 清理缓存
    unset($this->actionsSorted[$hook]);
}